core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StreamBatchSource.java

```diff
@@ -94,6 +94,10 @@
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     streamBatchConfig.validate();
     pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
+    // if no format is specified then default schema is used, if otherwise its based on input records.
+    if (streamBatchConfig.format == null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
+    }
```

Comment:
many formats will take a schema, which we can set as the outputschema.

Can use `streamBatchConfig.getFormatSpec()`, then if the spec's format is non-null, set the output schema accordingly.  One thing that is a little tricky is that the spec's schema is the schema of the body, so you'd have to add ts and headers as fields before setting the output schema.